automation/test.sh

```diff
@@ -43,14 +43,54 @@ else
 fi
 
 export KUBEVIRT_NUM_NODES=2
-export NFS_WINDOWS_DIR=${NFS_WINDOWS_DIR:-/home/nfs/images/windows2016}
+export WINDOWS_NFS_DIR=${WINDOWS_NFS_DIR:-/var/lib/stdci/shared/images/windows2016}
+export WINDOWS_LOCK_PATH=${WINDOWS_LOCK_PATH:-/tmp/download_windows_image.lock}
+
+wait_for_windows_lock() {
+  local max_lock_attempts=60
+  local lock_wait_interval=60
+
+  for ((i = 0; i < $max_lock_attempts; i++)); do
+      if (set -o noclobber; > $WINDOWS_LOCK_PATH) 2> /dev/null; then
+          echo "Acquired lock: $WINDOWS_LOCK_PATH"
+          return
+      fi
+      sleep $lock_wait_interval
+  done
+  echo "Timed out waiting for lock: $WINDOWS_LOCK_PATH" >&2
+  exit 1
+}
+
+release_windows_lock() {      
+  if [[ -e "$WINDOWS_LOCK_PATH" ]]; then
+      rm -f "$WINDOWS_LOCK_PATH"
+      echo "Released lock: $WINDOWS_LOCK_PATH"
+  fi
+}
+
+if [[ $TARGET =~ windows.* ]]; then
+  # Create images directory
+  if [[ ! -d $WINDOWS_NFS_DIR ]]; then
+    mkdir -p $WINDOWS_NFS_DIR
+  fi
+
+  # Download windows image
+  if wait_for_windows_lock; then
+    if [[ ! -f "$WINDOWS_NFS_DIR/disk.img" ]]; then
```

Comment:
What if we update the image? Is it a possible scenario? if so, maybe we can compare image checksum or something?

Reply:
I think we can add it in the future, if we will have a new image, currently not really relevant